Blaze on Woo Express: Adding new filter UI for woo products

ernestoSk13 opened the pull request:
Task:
Tumblr/a8c-dsp/issues/781

Related to #

Blaze on Woo Express

## Proposed Changes

* In this PR I'm adding the new UI for product filtering on the blaze dashboard for a Woo Store.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open calypso and select a Woo Store
* In the left panel select Marketing and then Blaze for Woo Commerce.
* You'll see the available products in the right. Look for the search bar and see that there's a Segmented Control with the options "Products" and "Posts & Pages"
* Toggle between the two options and make sure they are changing 


## Screenshots

Web:

<img width="1124" alt="Screenshot 2024-01-10 at 17 54 51" src="https://github.com/Automattic/wp-calypso/assets/5014402/0b6a66f9-85fc-4186-ba5e-21935753d24f">


Mobile:

![Screenshot 2024-01-10 at 2 19 10 p m](https://github.com/Automattic/wp-calypso/assets/5014402/734f6d73-76e7-46a7-82b8-794875ff1cbf)

Empty State message (Translators):

![Screenshot 2024-01-15 at 11 19 38 a m](https://github.com/Automattic/wp-calypso/assets/5014402/f5e11b20-11e0-4388-b8ad-54d350224b80)




## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
Adding new filter UI for woo products

Commit:
Adding new flag and making new filter UI functional

Commit:
Making filters UI functional

Commit:
Merge branch 'trunk' of https://github.com/Automattic/wp-calypso into add/woo-express-blaze-781

Commit:
Improving code and unifying woo and blog's postType

therocket-gr reviewed (commented):
I can see the new header of the table ok 
(one minor note is that the focus state has a stroke that maybe we should remove - ss attached below)
 
![Screenshot 2024-01-09 at 14 12 30](https://github.com/Automattic/wp-calypso/assets/1416426/815c929d-3ce2-437c-a811-19c42a83382a)

I can see the network call when i click the products button (and i think it is correct)

![Screenshot 2024-01-09 at 14 21 03](https://github.com/Automattic/wp-calypso/assets/1416426/172c4293-d11a-4d15-b0c1-6c60a6e03ee6)

but i have created a product item in my demo site and it comes empty
![Screenshot 2024-01-09 at 14 23 03](https://github.com/Automattic/wp-calypso/assets/1416426/f675f238-88df-4334-ad00-be0b7f8403f6)

![Screenshot 2024-01-09 at 14 23 56](https://github.com/Automattic/wp-calypso/assets/1416426/45f736fd-e45b-4fef-9da0-9f47c90aba1b)



![Screenshot 2024-01-09 at 14 22 50](https://github.com/Automattic/wp-calypso/assets/1416426/0d8f79d2-54d1-400e-a7fc-9fe3d986d561)

sbarbosa commented on client/my-sites/promote-post-i2/components/search-bar/index.tsx:
```diff
@@ -92,15 +98,31 @@ export default function SearchBar( props: Props ) {
+			label: translate( 'Posts' ),
+		},
+		{
+			value: 'pages',
```
I think this should be `page` (singular). It isn't filtering Ok right now.

ernestoSk13 replied:
let me correct this :)

ernestoSk13 replied:
Solved

sbarbosa commented on client/my-sites/promote-post-i2/components/search-bar/index.tsx:
```diff
@@ -199,15 +247,20 @@ export default function SearchBar( props: Props ) {
 				{ mode === 'posts' && (
 					<>
+						{ isWooStore && isDesktop && (
+							<WooItemsFilter handleChangeFilter={ onChangeFilter } postType={ postType || '' } />
```
Can't we use the same handler `onChangePostTypeFilter` as the default one here? 
Maybe it gets a bit messy because both will need to use the same value type `DropdownOption`. but right now the code is almost the same, so it would be nice to have a single dropdown.

ernestoSk13 replied:
I'm refactoring this to use only one method :)

ernestoSk13 replied:
Solved

sbarbosa reviewed (commented):
Hi @ernestoSk13. I tested it and just like @therocket-gr mentioned before, the filer doesn't appear to be working OK.
For example, for my in production I can filter by pages, but now it appears to not be working (probably a wrong constant).

Other than that, I found a couple of small bugs:

1- There is no separation between the new filter and the order by dropdown. Also in the design, the search bar appears to be smaller, and the order by drowdown didn't have a border.

![Screenshot 2024-01-09 at 11 17 12](https://github.com/Automattic/wp-calypso/assets/1258162/21683484-81ac-4ed6-88ea-66840ff39362)

![Screenshot 2024-01-09 at 11 26 03](https://github.com/Automattic/wp-calypso/assets/1258162/565a9adc-ef9a-4324-99de-1662bd64f9ec)

2- In tablet view, the filter is getting getting pushed down to the second row. I think for this view we also need to show the dropdown version of the filter.
![Screenshot 2024-01-09 at 11 17 55](https://github.com/Automattic/wp-calypso/assets/1258162/7f87034d-077c-4bf0-8bb0-ff7a0d3ee0f6)

3- When you first load the page we are making two request to the `/blaze/posts` endpoint. This is probably because we are not setting the correct default value for the filter. (for woo stores, the default is not blank, is products).

![Screenshot 2024-01-09 at 11 19 08](https://github.com/Automattic/wp-calypso/assets/1258162/0605f632-69c1-4a5b-8d37-3449f7e57987)

Commit:
Fixing UI issues

Commit:
Merge branch 'trunk' into add/woo-express-blaze-781

Commit:
Fixing UI issues on filters UI

ernestoSk13 commented:
@therocket-gr  @sbarbosa  the PR is ready to be reviewed again :) I have addressed the comments. For mobile I'm still using a dropdown until Bart let us know how to position this segmented control

Commit:
Fixing UI issue: wrong spacing between dropdown and segmented control

therocket-gr commented:
i see that now the items are working better (when i click only pages i get only pages, and products seem to work)

but.. (i will add some screenshots as it is easier to explain the issues i am seeing) 

1. i click on posts and only posts are displaying
![Screenshot 2024-01-11 at 14 21 22](https://github.com/Automattic/wp-calypso/assets/1416426/e828ecbe-fb97-4e4f-9a06-fd2fe031cc32)

2. i then click on the dropdown and select some other setting (most liked in this case.. but i tested and it has the same behaviour with other settings) the screen will load content and will come back with mixed posts/pages/products 
![Screenshot 2024-01-11 at 14 21 31](https://github.com/Automattic/wp-calypso/assets/1416426/f3a9153f-a74c-4d09-a991-5e5ff99452e1)
even though i only have posts selected.. same happens with other settings from the drop down menu
![Screenshot 2024-01-11 at 14 21 46](https://github.com/Automattic/wp-calypso/assets/1416426/da004c92-716e-4e24-a818-e392afc29c31)

sbarbosa commented:
Hi @therocket-gr, did you try it with the last change? I just tested it, and it worked OK for me. And checking your screenshot, they show an outdated version of the styles (the Sort field doesn't have the border now).

@ernestoSk13, I checked and I am still able to reproduce these two problems:
1- The tablet view shows the tabs filter, instead of the dropdown. If we are going for the dropdown version in mobile, I think it will be better to use that version also for tablet. (In prod works like that).
![Screenshot 2024-01-11 at 10 21 59](https://github.com/Automattic/wp-calypso/assets/1258162/1dbb152e-42fd-494c-a2ab-af6d40053bab)
2- The default value of the filters is not loaded correctly and it is making the dashboard to fetch twice for post on page load.
![Screenshot 2024-01-11 at 10 22 57](https://github.com/Automattic/wp-calypso/assets/1258162/a04d07a3-b746-4c35-b567-3f003e1874b4)


Also, I find some strange behavior with the Post type label for mobile. If you load the normal version of the page (no feature flag) and shrink the page, you will notice that we lost the Post type label when we get to the mobile screen size. It doesn't happen in prod.
![Screenshot 2024-01-11 at 10 27 49](https://github.com/Automattic/wp-calypso/assets/1258162/f7e7af38-a755-4f5d-8e9a-43c0b560031b)

therocket-gr commented:
> did you try it with the last change?

hm i think my git is stuck in a limbo.. maybe it wasnt updated (i hit pull but i think it didnt pulled everything) will try again and let you know

therocket-gr commented:
@ernestoSk13 yes it seems to be working fine now.. please ignore my previous comment... and sorry about that

Commit:
Fixing Github comments for filter UI

Commit:
Merge branch 'add/woo-express-blaze-781' of https://github.com/Automattic/wp-calypso into add/woo-express-blaze-781

Commit:
Merge branch 'trunk' into add/woo-express-blaze-781

ernestoSk13 commented:
hi @sbarbosa  @therocket-gr  I'm ready for another review :) Just one thing... do you know how to see that "E2E Tests" on TeamCity? I keep having those two checks failing

therocket-gr commented:
> hi @sbarbosa @therocket-gr I'm ready for another review :) Just one thing... do you know how to see that "E2E Tests" on TeamCity? I keep having those two checks failing

i rerun the teamcity tests.. now all tests are passing :) 
you can click details (next to the failed tests, login in team city and rerun the tests) 
i have also noticed that E2E Tests (mobile) and E2E tests (desktop) fail usually in some of my PRs, but i think they are just a bit flaky

sbarbosa reviewed (approved):
Thanks for the changes @ernestoSk13! 
The code looks great and it tested OK 👍

The only thing missing I think is an screenshot of the message we changed. It is usually very helpful for translators if we append a screenshot of the messages we change in our PR.

For this one, we only changed the message for Empty content.
Could you add an screenshot of that page?

therocket-gr reviewed (approved):
changes LGTM 
tested and it seems to be working fine now

Commit:
Merge branch 'trunk' into add/woo-express-blaze-781

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10819953

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @ernestoSk13 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10819953).
